### DISCUSSION TITLE
Mp3 files not working in Angular on Safari/iOS?

### DISCUSSION THREAD

**Discussion by Redn4s** [katspaugh/wavesurfer.js#3271] [Q&A]
Hi!

We are using wavesurfer.js in our Angular application. Recently we've updated from wavesurfer.js v6.6.4 to v7.3.8. Since then some mp3 files aren't working anymore on Safari/iOS. When calling the `playPause()` method we are encountering following error in the console:

`[Error] ERROR – Error: Uncaught (in promise): NotSupportedError: The operation is not supported.`

This error was introduced since v7 of wavesurfer.js. 

Below you can find a CodeSandox with a wavesurfer.js implementation in Angular. Running the sandbox in Safari and clicking the play button, will result in the same error shown above.

Do you know how we can make this wavesurfer.js v7 implementation work on Safari/iOS?

**Dependencies**
Wavesurfer.js: v7.3.8
Angular: v16.2.9
Safari: v16.6

**Demo**
CodeSandbox: https://codesandbox.io/p/sandbox/angular-wavesurfer-82zkkj

**Comment by katspaugh** [comment]
It's something with the file format, Safari cannot play it.

https://wavesurfer.xyz/examples/?287d9cb11765ab0229f7d03bd89e84c9 – getting the same error in this simplified example.

https://wavesurfer.xyz/examples/?aecce17467dc1196adea20942eefc1e8 – the same with a regular `<audio>` element.

**Reply by Redn4s** [reply]
I don't know if it's completely related to the file format, because the same mp3 file was working perfectly on v6.6.4 of wavesurfer.js.

I quickly duplicated the previous CodeSandbox and changed wavesurfer.js v7.3.8 to v6.6.4. 

CodeSandbox: https://codesandbox.io/p/sandbox/angular-wavesurfer-forked-5jrj87

If you run this sandbox, you'll notice that the same mp3 file is now working. 🙂

**Reply by katspaugh** [reply]
v6 used to play audio via a Web Audio buffer, v7 only supports HTML5 audio element playback. I'm planning to add Web Audio again in near future.
